webhooks-extension/src/components/WebhookBranches/WebhookBranches.js

```diff
@@ -0,0 +1,151 @@
+import React, { Component } from "react";
+import { Modal } from "carbon-components-react";
+import { fetchBranches } from "../../api";
+
+import {
+  DataTable,
+  DataTableSkeleton,
+  InlineNotification
+} from "carbon-components-react";
+
+const {
+  TableContainer,
+  Table,
+  TableHead,
+  TableRow,
+  TableBody,
+  TableCell,
+  TableHeader
+} = DataTable;
+
+export class WebhookBranches extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      rows: [],
+      loading: false,
+      error: null
+    };
+  }
+
+  componentDidMount() {
+    this.setState({
+      loading: true
+    });
+    let [, user, repo] = this.props.webhook
+      .toLowerCase()
+      .replace(/https?:\/\//, "")
+      .split("/");
+
+    fetchBranches(user, repo)
+      .then(branches => {
+        this.setState({
+          rows: branches.map(branch => ({
+            id: `${branch.name}-branch`,
+            branch: branch.name,
+            time: "No Results Available",
+            status: "No Results Available"
+          })),
+          loading: false
+        });
+      })
+      .catch(error => {
+        error.response.text().then(text => {
+          this.setState({
+            error: text,
+            rows: [
+              {
+                id: "branch",
+                branch: "-",
+                time: "-",
+                status: "-"
+              }
+            ],
+            loading: false
+          });
+        });
+      });
+  }
+
+  render() {
+    const { close } = this.props;
+    const { rows, loading, error } = this.state;
+
+    const headers = [
+      {
+        key: "branch",
+        header: "Branch"
+      },
+      {
+        key: "time",
+        header: "Last Build Time"
+      },
+      {
+        key: "status",
+        header: "Status"
+      }
+    ];
+
+    return (
+      <Modal
+        open={true}
+        id="webhook-branches-modal"
+        data-testid="webhook-branches-modal"
+        modalHeading="Branches"
+        passiveModal
+        onRequestClose={close}
+      >
+        {error !== null && (
+          <InlineNotification
+            data-testid="branches-modal-notification"
+            kind="error"
+            subtitle={error}
+            title="Error: "
+            lowContrast
+          />
+        )}
+        <DataTable
+          rows={rows}
+          headers={headers}
+          render={({ rows, headers, getHeaderProps, getRowProps }) => (
+            <TableContainer>
+              {loading ? (
+                <DataTableSkeleton
+                  rowCount={1}
+                  columnCount={headers.length}
+                  data-testid="loading-table"
+                />
+              ) : (
+                <Table className="bx--data-table--zebra">
```

Comment:
should this be using the DataTable prop to enable zebra styling instead of adding the class here? If Carbon changes the class name this would break, whereas the prop is part of the public API.

Reply:
still needs to be addressed ☝️